### PR TITLE
Refactor e2e collection helpers and commands to TS

### PR DESCRIPTION
**Pull request by kamilmielnik**
### Description

Will be needed in #45570

Refactors to TS:
- `cy.archiveCollection`
- `cy.createCollection`
- `e2e/support/helpers/e2e-collection-helpers.ts`
- `e2e/support/helpers/e2e-users-helpers.ts`

Also creates `archiveCollection` and `createCollection` function helpers which deprecate `cy.archiveCollection` and `cy.createCollection`.

### CODE REVIEW

File: e2e/support/helpers/e2e-users-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Get user's full name, or an email address if name is not available.
+ */
+export function getFullName(user: User): string {
+  const { first_name, last_name, email } = user;
+
+  if (first_name && last_name) {
+    return `${first_name} ${last_name}`;
+  }
+
+  if (first_name) {
```
Comment (kamilmielnik) [inline e2e/support/helpers/e2e-users-helpers.ts:16]: The body of this function changed a bit due to TS but it should work the same

File: e2e/support/helpers/index.js [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
+export * from "./e2e-slack-helpers";
+export * from "./e2e-snowplow-helpers";
+export * from "./e2e-ui-elements-helpers";
 export * from "./e2e-ui-elements-overflow-helpers";
-export * from "./e2e-relative-date-picker-helpers";
-export * from "./e2e-command-palette-helpers";
 export * from "./e2e-upload-helpers";
+export * from "./e2e-users-helpers";
+export * from "./e2e-visual-tests-helpers";
+export * from "./e2e-viz-settings-helpers";
```
Comment (kamilmielnik) [inline e2e/support/helpers/index.js:39]: Sorting export alphabetically

File: e2e/test/scenarios/organization/bookmarks-collection.cy.spec.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 import {
-  restore,
-  popover,
+  getSidebarSectionTitle,
   navigationSidebar,
+  popover,
+  restore,
   visitCollection,
 } from "e2e/support/helpers";
-import { getSidebarSectionTitle as getSectionTitle } from "e2e/support/helpers/e2e-collection-helpers";
```
Comment (kamilmielnik) [inline e2e/test/scenarios/organization/bookmarks-collection.cy.spec.js:12]: This rename was unnecessary

File: e2e/test/scenarios/organization/bookmarks-question.cy.spec.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-  restore,
+  getSidebarSectionTitle,
   navigationSidebar,
-  openQuestionActions,
   openNavigationSidebar,
+  openQuestionActions,
+  restore,
   visitQuestion,
 } from "e2e/support/helpers";
-import { getSidebarSectionTitle as getSectionTitle } from "e2e/support/helpers/e2e-collection-helpers";
```
Comment (kamilmielnik) [inline e2e/test/scenarios/organization/bookmarks-question.cy.spec.js:9]: This rename was unnecessary

File: e2e/support/helpers/e2e-collection-helpers.ts [hunk 5]
```diff
@@ -5,12 +5,43 @@ import {
...
+
+export const createCollection = ({
+  name,
+  description = null,
+  parent_id = null,
+  authority_level = null,
+}: {
+  name: string;
+  description?: string | null;
+  parent_id?: CollectionId | null;
```
Comment (nemanjaglumac) [inline e2e/support/helpers/e2e-collection-helpers.ts:18]: This is not strictly true.
`CollectionId` is one of the most misleading types we have. `parent_id` cannot be "personal", "users", or "trash". I'm almost 100% positive it cannot even be "root". So only `RegularCollectionId | null` seems to be correct here.

File: e2e/support/helpers/e2e-collection-helpers.ts [hunk 6]
```diff
@@ -5,12 +5,43 @@ import {
...
+  });
+};
 
 /**
  * Clicks the "+" icon on the collection page and selects one of the menu options
- * @param {"question" | "dashboard" | "collection"} type
  */
-export function openNewCollectionItemFlowFor(type) {
+export function openNewCollectionItemFlowFor(
+  type: "question" | "dashboard" | "collection",
```
Comment (nemanjaglumac) [inline e2e/support/helpers/e2e-collection-helpers.ts:43]: Should we add models and metrics to the list?

**Review by nemanjaglumac — APPROVED** [review]
Thank you

File: e2e/support/helpers/e2e-collection-helpers.ts (hunk 5 above)
Comment (kamilmielnik) [reply e2e/support/helpers/e2e-collection-helpers.ts]: 5ed7ad7a8c3ae45dd4e90fc10e6abd34fbba4172

File: e2e/support/helpers/e2e-collection-helpers.ts (hunk 6 above)
Comment (kamilmielnik) [reply e2e/support/helpers/e2e-collection-helpers.ts:49]: It's only ever used for "collections" at the moment.
Actually `newButton` helper should be used instead.

So I deprecated this function and I'm leaving the implementation as is: eb9a98020ae59a49a4d7388944025b4aad353903

**Pull request merged**